framework/src/meshgenerators/CircularCorrectionGenerator.C

```diff
@@ -0,0 +1,417 @@
+                                                (pt_rad - bdry_rad + transition_layer_thickness) /
+                                                transition_layer_thickness)
+                             : (pt_rad >= bdry_rad + transition_layer_thickness
+                                    ? 1.0
+                                    : 1.0 + (corr_factor - 1.0) *
+                                                (bdry_rad + transition_layer_thickness - pt_rad) /
+                                                transition_layer_thickness);
+      if (!MooseUtils::absoluteFuzzyEqual(mod_corr_factor, 1.0))
+      {
+        if (!mod_node_list.emplace((*node).id()).second)
+          paramError("transition_layer_ratios", "the transition layers are overlapped.");
+      }
+      const Point tmp_pt_alt =
+          Point(pt_rad * std::cos(tmp_pt_azi_new), pt_rad * std::sin(tmp_pt_azi_new), tmp_pt(2)) *
+          mod_corr_factor;
+      (*node) = tmp_pt_alt + boundary_origin;
+    }
+  }
+
+  return dynamic_pointer_cast<MeshBase>(_input);
+}
+
+Point
+CircularCorrectionGenerator::circularCenterCalculator(const std::vector<Point> pts_list,
+                                                      Real & radius,
+                                                      const Real tol)
+{
+  // Usually, just using the first three points would work
+  // Here a more complex selection is made in case the first three points are too close to each
+  // other.
+  // Step 1: find the farthest point from the first point
+  Real tmp_dis(0.0);
+  unsigned int farthest_pt_id(0);
+  for (unsigned int i = 1; i < pts_list.size(); i++)
+    if ((pts_list[i] - pts_list[0]).norm() > tmp_dis)
+    {
+      tmp_dis = (pts_list[i] - pts_list[0]).norm();
+      farthest_pt_id = i;
+    }
+  // Step 2: find the third point that is nearly in the middle of the first two
+  unsigned int mid_pt_id(0);
+  for (unsigned int i = 1; i < pts_list.size(); i++)
+    if (i != farthest_pt_id && std::abs((pts_list[farthest_pt_id] - pts_list[i]).norm() -
+                                        (pts_list[0] - pts_list[i]).norm()) < tmp_dis)
+    {
+      tmp_dis = std::abs((pts_list[farthest_pt_id] - pts_list[i]).norm() -
+                         (pts_list[0] - pts_list[i]).norm());
+      mid_pt_id = i;
+    }
+
+  const Real x12 = pts_list[0](0) - pts_list[farthest_pt_id](0);
+  const Real x13 = pts_list[0](0) - pts_list[mid_pt_id](0);
+
+  const Real y12 = pts_list[0](1) - pts_list[farthest_pt_id](1);
+  const Real y13 = pts_list[0](1) - pts_list[mid_pt_id](1);
+
+  const Real sx13 = std::pow(pts_list[0](0), 2.0) - std::pow(pts_list[mid_pt_id](0), 2.0);
+  const Real sy13 = std::pow(pts_list[0](1), 2.0) - std::pow(pts_list[mid_pt_id](1), 2.0);
+
+  const Real sx21 = std::pow(pts_list[farthest_pt_id](0), 2.0) - std::pow(pts_list[0](0), 2.0);
+  const Real sy21 = std::pow(pts_list[farthest_pt_id](1), 2.0) - std::pow(pts_list[0](1), 2.0);
```

Comment:
This isn't an inner kernel so it's not important here, but in general just square numbers with `x*x`.  [Compilers can't always optimize `pow(x,2)` as well](https://stackoverflow.com/a/2940800), especially if you don't turn on `-ffast-math` (which you shouldn't, as it changes *every* floating point expression's interpretation from IEEE 754 to YOLO 666).  I suspect using `2.0` and so instructing the compiler not to use its `pow(double,int)` overload makes their job even harder.

There's also a `libMesh::pow<N>(x)` that scales well for small fixed integer N, if you want that for readability (or for making sure the compiler doesn't repeat `operator[]` or `operator()` redundantly).